x/concentrated-liquidity/math.go

```diff
@@ -33,48 +33,59 @@ func liquidity1(amount sdk.Int, sqrtPriceA, sqrtPriceB sdk.Dec) sdk.Dec {
 // sqrtPriceA is the smaller of sqrtpCur and the nextPrice
 // sqrtPriceB is the larger of sqrtpCur and the nextPrice
 // calcAmount0Delta = (liquidity * (sqrtPriceB - sqrtPriceA)) / (sqrtPriceB * sqrtPriceA)
-func calcAmount0Delta(liq, sqrtPriceA, sqrtPriceB sdk.Dec) sdk.Dec {
+func calcAmount0Delta(liq, sqrtPriceA, sqrtPriceB sdk.Dec, roundUp bool) sdk.Dec {
 	if sqrtPriceA.GT(sqrtPriceB) {
 		sqrtPriceA, sqrtPriceB = sqrtPriceB, sqrtPriceA
 	}
 	diff := sqrtPriceB.Sub(sqrtPriceA)
 	mult := liq
+	// if calculating for amountIn, we round up
+	// if calculating for amountOut, we don't round
+	// this is to prevent removing more from the pool than expected due to rounding
```

Comment:
Could you elaborate a bit on this please? Why would we remove more than expected from pool? Where exactly the difference leads? For example, offline we discussed an infinite loop. Where is that loop?

Reply:
elaborated on this further. this should also address your concern in math.go. If it doesn't, I don't have any other way I can further explain this so will have to get on a call to further hash out what we can say to make it clearer.

Reply:
This is perfect, thank you